datalad/support/gitrepo.py

```diff
@@ -2201,10 +2208,12 @@ def _call_git(self, args, files=None, expect_stderr=False, expect_fail=False,
         expect_fail : bool, optional
           A non-zero exit is expected and should not be elevated above the
           DEBUG level.
-        check_fake_dates : bool, optional
-          If enabled (default), inspect if dates should be faked for dataset
-          commits and set up the Git environment to ensure that. Must be
-          disabled for repository initialization.
+        read_only : bool, optional
+          If set to True, we assume that command does not introduce changes to recorded
+          in git information (commits, heads, tags) and merely reads/outputs information
+          (note: git might still adjust caches and index).
+          If set to False, we would inspect if dates should be faked for dataset
+          commits and set up the Git environment to ensure that.
```

Comment:
How about something like this?  I'm trying to make it clearer that `read_only=True` isn't appropriate for the cases where git does an opportunistic refresh.

```suggestion
          By setting this to True, the caller indicates that the command does
          not write to the repository, which lets this function skip some
          operations that are necessary only for commands the modify the
          repository. Beware that even commands that are conceptually
          read-only, such as `git-status` and `git-diff`, may refresh and write
          the index.
```